src/infer_request.cc

```diff
@@ -173,10 +177,10 @@ InferRequest::SaveToSharedMemory(std::unique_ptr<SharedMemoryManager>& shm_pool)
   requested_output_names_shm_ = std::move(requested_output_names_shm);
 }
 
-std::unique_ptr<InferRequest>
+std::shared_ptr<InferRequest>
```

Comment:
The reason we return a shared pointer for the infer request object is that in the `async_exec` call we want to keep the infer request object around until the request execution is finished. Because of this the callback needs to keep a reference to the infer request object until it is the request's turn to execute. This is done in the L722 of pb_stub.cc.